[DOCS-858] Adding log processing rule examples for docker and K8s (#7273)
DataDog/documentation: content/en/agent/docker/log.md

- Use the environment variable to collect logs from all containers and then override the default `source` and `service` values.
- Add processing rules for the wanted subset of containers.

## Advanced log collection

> Use the autodiscovery log label to apply an advanced log collection processing logic like:

Reviewer: ```suggestion
Use Autodiscovery log labels to apply advanced log collection processing logic, for example:
```
